src/mpi/comm/comm_split_type.c

```diff
@@ -253,67 +222,37 @@ int MPIR_Comm_split_type_node_topo(MPIR_Comm * user_comm_ptr, int split_type, in
 
     /* initially point to HWLOC_OBJ_TYPE_MAX and then see if there is
      * an info argument pointing to a different object */
-    info_obj.obj_type = HWLOC_OBJ_TYPE_MAX;
+    obj_type = HWLOC_OBJ_TYPE_MAX;
     if (info_ptr) {
         MPIR_Info_get_impl(info_ptr, SHMEM_INFO_KEY, MPI_MAX_INFO_VAL, hintval, &flag);
         if (flag) {
             /* try to find the info value in the processor object
              * table */
             for (i = 0; shmem_processor_info[i].val; i++) {
                 if (!strcmp(shmem_processor_info[i].val, hintval)) {
-                    info_obj.obj_type = shmem_processor_info[i].obj_type;
+                    obj_type = shmem_processor_info[i].obj_type;
                     break;
                 }
             }
-
-            /* if we have not found it yet, try to see if the value
-             * matches any of our known strings.  right now, we only
-             * understand "cache". */
-            if (info_obj.obj_type == HWLOC_OBJ_TYPE_MAX) {
-                if (!strncmp(hintval, "cache:l", strlen("cache:l"))) {
-                    const char *d = hintval + strlen("cache:l");
-                    info_obj.u.cache.depth = 0;
-                    while (*d >= '0' && *d <= '9') {
-                        info_obj.u.cache.depth = 10 * info_obj.u.cache.depth + *d - '0';
-                        d++;
-                        if (*d == '\0')
-                            break;
-                    }
-
-                    if (info_obj.u.cache.depth == 0)
-                        goto use_node_comm;
-
-                    info_obj.obj_type = HWLOC_OBJ_CACHE;
-                    if (*d == 'u')
-                        info_obj.u.cache.cache_type = HWLOC_OBJ_CACHE_UNIFIED;
-                    else if (*d == 'i')
-                        info_obj.u.cache.cache_type = HWLOC_OBJ_CACHE_INSTRUCTION;
-                    else if (*d == 'd' || *d == '\0')
-                        info_obj.u.cache.cache_type = HWLOC_OBJ_CACHE_DATA;
-                    else
-                        goto use_node_comm;
-                }
-            }
```

Comment:
It's OK to remove this for now, but once we add the I/O devices, we'll need to revive some part of this code to deal with them.